pkg/kubelet/server/stats/summary.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stats
+
+import (
+	"fmt"
+
+	cadvisorapiv2 "github.com/google/cadvisor/info/v2"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+// Handles stats summary requests to /stats/summary
+func buildSummary(
+	node *api.Node,
+	pods []*api.Pod,
+	infos map[string]cadvisorapiv2.ContainerInfo) (*Summary, error) {
+
+	rootInfo, found := infos["/"]
+	if !found || len(rootInfo.Stats) < 1 || rootInfo.Stats[0] == nil {
+		return nil, fmt.Errorf("Missing stats for root container")
+	}
+
+	rootStats := containerInfoV2ToStats("", &rootInfo)
+	nodeStats := NodeStats{
+		NodeName: node.Name,
+		CPU:      rootStats.CPU,
+		Memory:   rootStats.Memory,
+		Network:  containerInfoV2ToNetworkStats(&rootInfo),
+	}
+
+	systemContainers := map[string]string{
```

Comment:
Can this pull the list of system containers from ContainerManager?  I worry this will get lost when we fix things up for systemd if not pulled from another source of truth.